crypto/params_idx.c.in

```diff
@@ -0,0 +1,39 @@
+/*
+ * {- join("\n * ", @autowarntext) -}
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+{-
+use OpenSSL::paramnames qw(procduce_decoder);
+-}
+
+#include "internal/e_os.h"
+#include "internal/param_names.h"
+#include <string.h>
+
+
+/* TRIE based decoder - generated by util/perl/OpenSSL/paramnames.pm */
+{- procduce_decoder(); -}
+/* End of TRIE */
+
+int ossl_scan_params_for_idx(const OSSL_PARAM *params, struct param_idx *idx)
+{
+    int n, c = 0;
+    unsigned char i;
+
+    for (i = 0; params[i].key != NULL; i++) {
+        n = ossl_param_find_idx(params[i].key);
+        if (n >= 0 && c < MAX_PARAM_COUNT - 1) {
```

Comment:
Shouldn't this function try to catch it when `c == MAX_PARAM_COUNT` and signal an error (possibly by returning -1...  or in this case, I'd even go as far as having an assertion, 'cause a too small `MAX_PARAM_COUNT` is an implementation error, is it not?)

Reply:
I meant to increment the counter even if it couldn't write.  Returning an error is also possible but less helpful.  The function could be called a second time with params+MAX_PARAM_COUNT-1 to process the rest.
I expect 100 parameters to be sufficient for everything except a denial of service attack :)